assets/javascripts/discourse/widgets/qa-button.js

```diff
@@ -7,7 +7,7 @@ export default createWidget("qa-button", {
   buildAttributes(attrs) {
     const attributes = {};
 
-    if (attrs.loading) {
+    if (attrs.loading || attrs.disabled) {
```

Comment:
When it is disabled, I think it’ll also be helpful UX wise to state why the voting button is disabled.

Reply:
Currently we are disabling the buttons on self answer/comments. In the future we'll also disable the buttons after 10 mins (`qa undo vote action window`) of voting. We'll be using a single attribute (`qa_can_vote`) to decide if the button should be disabled or not. So it will get tricky to add an exact explanation as to why the button is disabled.

In my next PR to disable the buttons based on `qa undo vote action window` I'll see if we can add an helpful title that can add reasoning for both the cases.

Reply:
Actually, I'll rather we avoid all the disabled logic and just return a proper error from the server side stating why the vote cannot be created. That would be the simplier solution IMO.

Reply:
That does sound a lot simpler to do. I was thinking of avoiding firing unnecessary AJAX requests by disabling the buttons. Is it okay if we trigger AJAX requests even on our own posts/comments in favour of showing better error messages?

Reply:
I think that is fine and is what stackoverflow does too.

Reply:
Okay, let me update the PR now to perform the check on server side and reply with better error messages.

Reply:
Done!